### ISSUE TITLE
WebSocket SSL error self-signed certificate

### ISSUE THREAD

**Issue by kaihendry** [vi/websocat#191]
I noticed https://github.com/vi/websocat/issues/122 though I don't understand why a self signed certificate with **mtls** should fail?

``` 
websocat: WebSocketError: WebSocket SSL error: error:0A000086:SSL routines:tls_post_process_server_certificate:certificate verify failed:ssl/statem/statem_clnt.c:1889: (self-signed certificate)
websocat: error running
```
-k is documented as "Accept invalid certificates and hostnames while connecting to TLS". There is nothing invalid here?

server moans like so:
```websocat: error:0A000418:SSL routines:ssl3_read_bytes:tlsv1 alert unknown ca:ssl/record/rec_layer_s3.c:1586:SSL alert number 48```

❯ openssl version
OpenSSL 3.1.1 30 May 2023 (Library: OpenSSL 3.1.1 30 May 2023)

https://gist.github.com/kaihendry/990027a7c9a54efcdf138d45f720088e

Thank you!

**Comment by vi** [comment]
Does some workaround with `websocat` spawning subprocess instead of doing TLS work?

> server moans like so: ...

Can you connect to server instance of Websocat from a browser (after temporarily whilelisting the domain for invalid cert)?

**Comment by kaihendry** [comment]
I get a `websocat: error:0A00009C:SSL routines:ssl3_get_record:http request:ssl/record/ssl3_record.c:345:` when I try point Chrome to http://localhost:4321/ 


[wss](wss://localhost:4321/) is surprisingly a ERR_UNKNOWN_URL_SCHEME

**Comment by kaihendry** [comment]
Yes it works, Chrome does complain about `NET::ERR_CERT_AUTHORITY_INVALID`, though I didn't expect websocat server to mediate the CA, unless I misunderstood.


![wss](https://github.com/vi/websocat/assets/765871/f39cb2e1-ea57-4d0c-bc4b-997359460e61)
